operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractExternalDependentResource.java

```diff
@@ -0,0 +1,120 @@
+package io.javaoperatorsdk.operator.processing.dependent;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.RecentOperationCacheFiller;
+import io.javaoperatorsdk.operator.processing.event.EventSourceRetriever;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.source.ResourceEventSource;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+
+public abstract class AbstractExternalDependentResource<R, P extends HasMetadata, T extends ResourceEventSource<R, P>>
+    extends AbstractEventSourceHolderDependentResource<R, P, T> {
+
+  private final boolean isDependentResourceWithExplicitState =
```

Comment:
I'm not sure renaming the class will fix the problem… Why would someone extend `AbstractExternalDependentResource` and not implement `DependentResourceWithExplicitState`? Could you detail a use case where someone would extend this class and not want an explicit external state with it?

Reply:
yep, so we have already samples for that (like the mysql sample):
If somebody want a PollingDependentResource, that does not need an external state.

Note that not every external resource needs an explicit state, for example if it can be addressed only by the information in the custom resource. However some might need one, like if a github PR is managed. Therefore the current implementation allows to a dependent resource to extend PollingDependentResource (and others) and to decide if it needs an external state or not implementing the related `DependentResourceWithExplicitState`.

I thing that we might address in a separate PR and could have an impact on this is this issue:
https://github.com/java-operator-sdk/java-operator-sdk/issues/1511
(but not necessarily)

However I recognize that Polling and PerResourcePolling DRs are quite a thin layer, that basically just creates an instance of a EventSource on top of a standard functionality. (althoug might be extended later for example with an improved configuration). So the structure might change in the future if we for example make it less explic? "Thus just an external event source that provides an event source" abstraction.

Reply:
Yep, but don't see how we would cover otherwise just optionally having this external state functionality for Polling Event sources.